doc/user/content/ingest-data/mysql/self-hosted.md

```diff
@@ -0,0 +1,154 @@
+`enforce_gtid_consistency`       | `ON`   |
+`binlog retention hours`         | 168    |
+`replica_preserve_commit_order`  | `ON`   | Only required when connecting Materialize to a read-replica for replication, rather than the primary server.
+
+For guidance on enabling GTID-based binlog replication, see the
+[MySQL documentation](https://dev.mysql.com/blog-archive/enabling-gtids-without-downtime-in-mysql-5-7-6/).
+
+## Step 2. Create a user for replication
+
+{{% mysql-direct/create-a-user-for-replication %}}
+
+## Step 3. Configure network security
+
+There are various ways to configure your database's network to allow Materialize
+to connect:
+
+- **Allow Materialize IPs:** If your database is publicly accessible, you can
+    configure your database's firewall to allow connections from a set of
+    static Materialize IP addresses.
+
+- **Use an SSH tunnel:** If your database is running in a private network, you
+    can use an SSH tunnel to connect Materialize to the database.
+
+Select the option that works best for you.
+
+{{< tabs >}}
+
+{{< tab "Allow Materialize IPs">}}
+
+1. In the [SQL Shell](https://console.materialize.com/), or your preferred SQL
+   client connected to Materialize, find the static egress IP addresses for the
+   Materialize region you are running in:
+
+    ```sql
+    SELECT * FROM mz_egress_ips;
+    ```
+
+1. Update your database firewall rules to allow traffic from each IP address
+   from the previous step.
+
+{{< /tab >}}
+
+{{< tab "Use an SSH tunnel">}}
+
+To create an SSH tunnel from Materialize to your database, you launch an VM to
+serve as an SSH bastion host, configure the bastion host to allow traffic only
+from Materialize, and then configure your database's private network to allow
+traffic from the bastion host.
+
+1. Launch a VM to serve as your SSH bastion host.
+
+    - Make sure the VM is publicly accessible and in the same VPC as your
+      database.
+    - Add a key pair and note the username. You'll use this username when
+      connecting Materialize to your bastion host.
+    - Make sure the VM has a static public IP address. You'll use this IP
+      address when connecting Materialize to your bastion host.
+
+1. Configure the SSH bastion host to allow traffic only from Materialize.
+
+    1. In the [SQL Shell](https://console.materialize.com/), or your preferred
+       SQL client connected to Materialize, get the static egress IP addresses for
+       the Materialize region you are running in:
+
+       ```sql
+       SELECT * FROM mz_egress_ips;
+       ```
+
+    1. Update your SSH bastion host's firewall rules to allow traffic from each
+       IP address from the previous step.
+
+1. Update your database firewall rules to allow traffic from the SSH bastion
+   host.
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+## Step 4. (Optional) Create a cluster
+
+{{< note >}}
+If you are prototyping and already have a cluster to host your MySQL
```

Comment:
Also wondering here and in other places if we should duplicate the text or write it once and include it (where possible).

Reply:
Yeah, there's a bunch of cleanup to do in the guides (also for PostgreSQL). We can definitely reduce the duplication of content across them.